server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -1244,16 +1244,16 @@ public void testStrictParsing() {
         assertParseException("2018-12-31T12:12:12", "strict_date_hour_minute_second_millis", 19);
         assertParseException("2018-12-31T12:12:12", "strict_date_hour_minute_second_fraction", 19);
         assertParses("2018-12-31", "strict_date_optional_time");
-        assertParseException("2018-12-1", "strict_date_optional_time", 7);
```

Comment:
The difference is because the ISO parser reports the offset at the start of the field (even if the field is partially there), but java.time reports the offset where there should be more data but isn't